main.py

```diff
@@ -51,11 +71,43 @@ def get_production_server(self) -> flask.Flask:
     def run_development_server(self) -> None:
         self._app.run_server(debug=True)
 
-    def _make_layout(self) -> html.Div:
+    def _get_page_content(self, pathname: str) -> html.Div:
+        try:
+            if pathname == "/":
+                return self._make_matrix_layout()
+            elif pathname.startswith("/chart"):
```

Comment:
This specific example won't work for us as chart view takes parameters:  /chart?from=343&to=565.
I will update the code to prevent "charting" case.
Agree that implementing real path routing is overkill at this stage and probably won't be necessary.

Reply:
I've assumed that the `pathname` contains only URL path without query parameters. If it includes parameters we can do it this way:
```
components = pathname.split("?")[0].split("/')
if len(components) == 0:
    return self._make_matrix_layout()
if components[1] == "chart":
    return self._make_chart_layout(pathname)
return HTTPErrorView.make_layout(404)
```

Reply:
I will wrap this eventually into separate tool-function